tests/src/tools/ReadyToRun.SuperIlc/CompilerRunner.cs

```diff
@@ -61,6 +67,59 @@ public virtual ProcessInfo CompilationProcess(string outputRoot, string assembly
             return processInfo;
         }
 
+        public IEnumerable<ProcessInfo> CompilationR2RDumpProcesses(string compiledExecutable)
+        {
+            if (_options.R2RDumpPath != null)
+            {
+                StringBuilder commonBuilder = new StringBuilder();
+
+                commonBuilder.Append($@"""{_options.R2RDumpPath.FullName}""");
+
+                commonBuilder.Append(" --normalize");
+                commonBuilder.Append(" --sc");
+                commonBuilder.Append(" --disasm");
+
+                foreach (string referencePath in _options.ReferencePaths())
+                {
+                    commonBuilder.Append($@" --rp ""{referencePath}""");
+                }
+                commonBuilder.Append($@" --in ""{compiledExecutable}""");
+
+                StringBuilder rawBuilder = new StringBuilder(commonBuilder.ToString());
+                rawBuilder.Append(" --naked");
+                string rawOutputFileName = compiledExecutable + ".raw.r2r";
+                rawBuilder.Append($@" --out ""{rawOutputFileName}""");
+
+                ProcessInfo rawInfo = new ProcessInfo();
+                rawInfo.ProcessPath = "dotnet";
```

Comment:
Hmm, I'm worried this is not going to work cross-plat. Could someone please advise me how to code this up properly?

Reply:
If I was doing this, I would structure r2rdump so that it can be used as a library and used it as a library.

Starting new processes gives me goosebumps (it's harder to debug the nested processes and in general many things can go wrong, especially if you also need to re-invoke `dotnet` that might end up being a different `dotnet` from the one running your tool).

Reply:
Well, that's the longer-term plan; sadly the fact that currently the two tools reside in different repos (CoreRT vs. CoreCLR) makes this quite involved. In fact, today it would be much easier to refactor the IL Compiler itself to be used as a statically linked-in module rather than as an external process by SuperILC - would you find such an option worth pursuing?

Reply:
We could factor the source in r2rdump so it has the front-end dumper that uses a set of classes that provide the dumping / diffing behavior we want to share. We could just copy those source files into this repo until we merge one into the other, or do the work to publish r2rdump as a dotnet tool. Probably a matter of opinion on which feels less skeezy 😄

Reply:
> IL Compiler itself to be used as a statically linked-in module rather than as an external process by SuperILC - would you find such an option worth pursuing?

SuperILC is a test tool. It should be invoking the compiler same way as the shipping config. I do not think that this is worth pursuing.

Reply:
At some point, hopefully soon, we should figure out how to publish CPAOT for the purpose of testing in the CoreCLR branch. We could also use this opportunity to stitch the two together. I think it's mostly trivial to split the app into a "front-end" and a "back-end library" but it's just more work - just like a balloon, the CPAOT budget cannot be inflated indefinitely, otherwise it pops at some point.

Reply:
Backed up by my 1:1 discussion with Sergiy I believe we should leave things as they are until the CPAOT compiler gets ported to the CoreCLR repo, then we can make the R2RDump library concept work.